### DISCUSSION TITLE
Conditional Navbar Link Rendering + React Callback Functions

### DISCUSSION THREAD

**Discussion by torinmb** [shuding/nextra#2799] [Q&A]
Hey, is there any way to conditionally show/ hide the navbar links? I'm also curious if instead of passing just an href to a link I can attach a react function to one of the links. (Also I know you can show / hide links by specifying them in the _meta.json. I have some state in react that I need to conditionally render certain links)

Using the theme.config.jsx I've tried implementing the navbar.component, but it replaces the whole nav and I want to keep the default navbar and build on top of it. 

I noticed if you implement the sidebar.titleComponent the style stays the same and you can add functionality to the sidebar, which is perfect. I'm curious if there's a parallel for the navbar. Any pointers would be greatly appreciated.

**Comment by kidfribble** [comment]
Adding to this! I've got an implementation using Clerk for auth and I'm trying to figure out how to hide the sign in/ sign up links from the nav bar when my users are signed in. Any help would be amazing!
